### PR TITLE
End to end testing: Fixture moved to a separate file

### PR DESCRIPTION
**Pull request by noopurintel**
Changes done:

1. Fixture `fx_federation` moved out of `conftest.py` to newly created `common_fixtures.py` under `tests/end_to_end/utils` to keep conftest file short and simple.
2. Correction made to docker workflow for OpenFL image creation step.
3. Correction made to task runner workflow for memory logs.
4. `README.md` updated to indicate that utils contain fixture file.

Successful jobs:

1. Task Runner workflow - https://github.com/noopurintel/openfl/actions/runs/12292745031
2. Task Runner via Docker workflow - https://github.com/noopurintel/openfl/actions/runs/12292746636

### CODE REVIEW

File: .github/workflows/task_runner_docker_e2e.yml [hunk 1]
```diff
@@ -66,6 +66,14 @@ jobs:
           pip install .
           pip install -r test-requirements.txt
 
+      - name: Create OpenFL image
+        id: create_openfl_image
+        run: |
+          echo "Creating openfl image. This may take a few minutes..."
```
Comment (payalcha) [inline .github/workflows/task_runner_docker_e2e.yml]: Creating openfl image with current branch and repository.

I think this comment will be more helpful.

**Review by payalcha — APPROVED** [review]

File: .github/workflows/task_runner_docker_e2e.yml (hunk 1 above)
Comment (noopurintel) [reply .github/workflows/task_runner_docker_e2e.yml]: Done.

**Review by rajithkrishnegowda — APPROVED** [review]

**Comment by rajithkrishnegowda** [comment]
TaskRunner (docker/gramine-direct) pipeline is failing. please check

**Comment by rajithkrishnegowda** [comment]
> TaskRunner (docker/gramine-direct) pipeline is failing. please check

on rerunning it worked

**Review by tanwarsh — APPROVED** [review]

**Pull request merged**